Sources/Commands/SwiftTestTool.swift

```diff
@@ -493,25 +493,26 @@ public struct SwiftTestTool: SwiftCommand {
     /// - Returns: Array of TestSuite
     fileprivate func getTestSuites(fromTestAt path: AbsolutePath, swiftTool: SwiftTool) throws -> [TestSuite] {
         // Run the correct tool.
-      #if os(macOS)
+        #if os(macOS)
         let data: String = try withTemporaryFile { tempFile in
             let args = [try xctestHelperPath(swiftTool: swiftTool).pathString, path.pathString, tempFile.path.pathString]
             var env = try constructTestEnvironment(toolchain: try swiftTool.getToolchain(), options: swiftOptions, buildParameters: swiftTool.buildParametersForTest())
             // Add the sdk platform path if we have it. If this is not present, we
             // might always end up failing.
             if let sdkPlatformFrameworksPath = Destination.sdkPlatformFrameworkPaths() {
-                env["DYLD_FRAMEWORK_PATH"] = sdkPlatformFrameworksPath.fwk.pathString
-                env["DYLD_LIBRARY_PATH"] = sdkPlatformFrameworksPath.lib.pathString
+                // appending since we prefer the user setting (if set) to the one we inject
+                env.appendPath("DYLD_FRAMEWORK_PATH", value: sdkPlatformFrameworksPath.fwk.pathString)
+                env.appendPath("DYLD_LIBRARY_PATH", value: sdkPlatformFrameworksPath.lib.pathString)
```

Comment:
is this enough or do we need to explicitly pick up and set the env variable from current env into this `env` FIRST and then add the SDK one...?

Reply:
var env = try constructTestEnvironment

Reply:
Nevermind, I see we do this in `    var env = EnvironmentVariables.process()` :)